### PR TITLE
Refactor #342 refactor internal user state as `locked` is basically a state transition

### LINKED ISSUE #342: refactor user lifecycle
### Description

the current user lifecycle implementation is used when activating a user after its registration. But there are further steps such as `locked` that are also valid, but should be part of the state rather than being put into a single bool variable.
#### Note

This is an architectural issue and nothing critical, so moved to 0.2.

### PR DESCRIPTION
**Pull request by Ma27**
### Overview

refactored user model to have better state transitions:

- modify behavior to handle the new state transitions properly
- refactor public API of the `User` aggregate
- improve coverage

### Related issues

resolves #342

### CODE REVIEW

File: src/AppBundle/EventListener/IncompleteUserCheckListener.php [hunk 1]
```diff
@@ -55,16 +55,16 @@ public function validateUserOnAuthentication(OnAuthenticationEvent $event)
...
 
         if ($isLocked || $isNonApproved || $this->temporaryBlockedAccountProvider->isAccountTemporaryBlocked($user->getId())) {
             switch (true) {
-                case $isNonApproved:
-                    $message = 'BACKEND_AUTH_NON_APPROVED';
-                    break;
                 case $isLocked:
                     $message = 'BACKEND_AUTH_LOCKED';
                     break;
+                case $isNonApproved:
```
Comment (Ma27) [inline src/AppBundle/EventListener/IncompleteUserCheckListener.php:69]: maybe add a notice here because of the change?

**Review by Ma27 — COMMENTED** [review]
:+1:

**Comment by Ma27** [comment]
@benbieler anything to add here?

**Comment by benbieler** [comment]
@Ma27 Sorry for not reviewing... Had to much work to do.

**Pull request merged**